maildir/shared.c

```diff
@@ -1700,17 +1700,8 @@ struct Account *maildir_ac_find(struct Account *a, const char *path)
  */
 int maildir_ac_add(struct Account *a, struct Mailbox *m)
 {
-  if (!a || !m)
+  if (!a || !m || m->magic != MUTT_MAILDIR)
```

Comment:
I'm reposting this since it appears GitHub did not want to link directly to the code:

The mh backend uses this function as well. We should expand the check to handle MUTT_MH.

It looks like this function had been broken since this guard clause was further down.

Reply:
Well catch, thx.